test/integration/integration_test.go

```diff
@@ -347,29 +289,42 @@ func TestDocsTopic(t *testing.T) {
 		},
 	} {
 		t.Run(name, func(t *testing.T) {
-			var repository *gitRepo
-			if c.kind == sourceGit {
-				repo, err := newGitRepository(t, addonSource)
-				assert.NoError(t, err)
-
-				defer repo.removeTmpDir()
-				repository = repo
-			}
-
-			// when
-			source := newSource(c.kind, suite, repository, []string{redisAndAccTestRepo})
-			suite.createClusterAddonsConfiguration(c.addonName, source)
+			suite.createClusterAddonsConfiguration(c.addonName, []string{redisAndAccTestRepo}, c.kind)
 
 			// then
 			suite.waitForClusterAddonsConfigurationPhase(c.addonName, v1alpha1.AddonsConfigurationReady)
 			suite.assertClusterDocsTopicExist(c.docsTopicID)
 
 			// when
-			source.replaceURL(redisRepo)
-			suite.updateClusterAddonsConfigurationRepositories(c.addonName, source)
+			suite.updateClusterAddonsConfigurationRepositories(c.addonName, []string{redisRepo}, c.kind)
 
 			// then
 			suite.assertClusterDocsTopicListIsEmpty()
 		})
 	}
 }
+
+func TestDisabledDocs(t *testing.T) {
+	suite := newTestSuite(t, false)
+	defer suite.tearDown()
+
+	t.Run("namespaced", func(t *testing.T) {
+		suite.assertNoServicesInCatalogEndpoint("ns/stage")
+
+		// when
+		suite.createAddonsConfiguration("stage", "addon1", []string{accTestRepo}, sourceHTTP)
+
+		// then
+		suite.waitForAddonsConfigurationPhase("stage", "addon1", v1alpha1.AddonsConfigurationReady)
+	})
+
+	t.Run("cluster", func(t *testing.T) {
+		suite.assertNoServicesInCatalogEndpoint("cluster")
+
+		// when
+		suite.createClusterAddonsConfiguration("addon1", []string{accTestRepo}, sourceHTTP)
+
+		// then
+		suite.waitForClusterAddonsConfigurationPhase("addon1", v1alpha1.AddonsConfigurationReady)
```

Comment:
please add `assertClusterDocsTopicListIsEmpty()` check for `namespaced` and `cluster` case

Reply:
CRDs are not installed so I cannot test it